server/routerlicious/packages/services/src/mongodb.ts

```diff
@@ -7,103 +7,174 @@ import { assert } from "console";
 import * as core from "@fluidframework/server-services-core";
 import { AggregationCursor, Collection, MongoClient, MongoClientOptions } from "mongodb";
 import { Lumberjack } from "@fluidframework/server-services-telemetry";
+import { requestWithRetry } from "@fluidframework/server-services-core";
+import { MongoErrorRetryAnalyzer } from "./mongoExceptionRetryRules";
 
 const MaxFetchSize = 2000;
 
-export class MongoCollection<T> implements core.ICollection<T> {
-    constructor(private readonly collection: Collection<T>) {
+export class MongoCollection<T> implements core.ICollection<T>, core.IRetryAble {
+    public readonly retryEnabled: boolean;
+    constructor(
+        private readonly collection: Collection<T>,
+        private readonly clientSideRetry?: boolean,
```

Comment:
Making this `public readonly retryEnabled: boolean = false` would be simpler than having both `retryEnabled` and `clientSideRetry`. Is there something I'm missing about a difference in their meanings?

Reply:
I think you made a validate commented. `retryEnabled` was added later as part of the `IRetryable` and introduced some sloppy code.

Reply:
Addressed in b0e2d37